portal/src/OnboardingSurveyScreen.tsx

```diff
@@ -723,8 +723,13 @@ function Step4(_props: StepProps) {
       e.stopPropagation();
       setLocalStorage("reasonChoices", reasonChoicesState);
       if (otherReason !== "") setLocalStorage("otherReason", otherReason);
+      const companyName = getFromLocalStorage("companyName");
       localStorage.removeItem(localStorageKey);
-      navigate("./../../projects/create");
+      if (companyName !== undefined)
+        navigate("./../../projects/create", {
+          state: { company_name: companyName },
```

Comment:
Use `replace: true` to replace the current history entry?

Reply:
Is it to prevent users from clicking back and resubmitting the survey? I tried `replace: true`, but the user is still able to go back to the survey; I think what is happening is that since submitting replaced step 4, pressing back returned to step 3, and since it detected localstorage is empty it redirects to step 1

Reply:
Yes you are right.
Ok, so we allow the user to back to the survey by using back, and it will start from step 1.